### DISCUSSION TITLE
classes/class_placeholdertexture2d

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#22] [User-contributed notes]
# classes/class_placeholdertexture2d

Inherits: Texture2D< Texture< Resource< RefCounted< Object Placeholder class for a 2-dimensional texture. Description: This class is used when loading a project that uses a Texture2D subclass in 2 ...

https://docs.godotengine.org/en/latest/classes/class_placeholdertexture2d.html

**Comment by ducklin5** [comment]
If you came here looking for an empty texture to use for shader based rendering, then you are better off using a different texture type entirely: `GradientTexture2D` or `NoiseTexture2D`. 

`PlaceholderTexture2D` was made for other purposes and it behaves differently in shaders
